client/my-sites/plans/main.jsx

```diff
@@ -265,7 +264,10 @@ class Plans extends Component {
 						{ isDomainAndPlanPackageFlow && (
 							<>
 								<div className="plans__header">
-									<DomainAndPlanPackageNavigation goBackLink={ goBackLink } step={ 2 } />
```

Comment:
This `goBackLink` is preventing a loop, where if you get all the way to checkout, then click the "X" to go back, then click the "< Back" button, without this prop it'll send you back to the shopping cart and put you in a loop between the plans page and the empty checkout page.

I'm guessing it was removed here so if the user clicks "Buy this domain" the Back button will send them Home and not to the Domain Search page?

Maybe we can dynamically set the `goBackLink` prop in case the user clicks "Buy this domain"? Alternatively, maybe it's ok if we just send them to the Domain Search page? They can get Home from that page, and it might entice them with a difference domain?

Reply:
I added back the goBackLink, but the problem now is if the user goes thru the My Home Upsell flow but searches for a domain, the back should be going back to domains on the plans page.